File: tests/snapshots/src/main/scala/tests/SnapshotHandler.scala
package tests

import java.nio.file.Path

abstract class SnapshotHandler {
  def onSnapshotTest(
      context: SnapshotContext,
      expectFile: Path,
      obtainedOutput: () => String
  ): Unit = ()
  def onFinished(context: SnapshotContext): Unit = ()
}

File: scip-snapshots/src/main/java/tests/SnapshotHandler.java
package tests;

import java.nio.file.Path;
import java.util.function.Supplier;

public abstract class SnapshotHandler {
  public void onSnapshotTest(
      SnapshotContext context, Path expectFile, Supplier<String> obtainedOutput) {}

  public void onFinished(SnapshotContext context) {}
}
